plugins/processors/gpuattributes/internal/awsneuron_memory_metric_aggregator.go

```diff
@@ -0,0 +1,88 @@
+	containerinsightscommon.NeuronCoreMemoryUtilizationRuntimeMemory:    {},
+	containerinsightscommon.NeuronCoreMemoryUtilizationTensors:          {},
+}
+
+type NeuronCoreInfo struct {
+	neuronCoreIndex   string
+	neuronDeviceIndex string
+}
+
+type AwsNeuronMemoryMetricsAggregator struct {
+	memoryMetricValuesAggregator     map[NeuronCoreInfo]float64
+	aggregatedMemoryMetricAttributes pcommon.Map
+	metricTimestamp                  pcommon.Timestamp
+	MemoryMetricsFound               bool
+}
+
+func NewMemoryMemoryAggregator() *AwsNeuronMemoryMetricsAggregator {
+	return &AwsNeuronMemoryMetricsAggregator{memoryMetricValuesAggregator: map[NeuronCoreInfo]float64{}, MemoryMetricsFound: false}
+}
+
+func (d *AwsNeuronMemoryMetricsAggregator) AggregateMemoryMetric(originalMetric pmetric.Metric) {
+	if _, exists := memoryMetricsNames[originalMetric.Name()]; exists {
+		datapoints := originalMetric.Gauge().DataPoints()
+		if datapoints.Len() > 0 {
+			d.MemoryMetricsFound = true
+			d.aggregatedMemoryMetricAttributes = datapoints.At(0).Attributes()
+			d.metricTimestamp = datapoints.At(0).Timestamp()
+
+			for i := 0; i < datapoints.Len(); i++ {
+				datapoint := datapoints.At(i)
+
+				neuronCoreIndexValue, neuronCoreIndexValueExists := datapoint.Attributes().Get(NeuronCoreAttributeKey)
+				neuronDeviceIndexValue, neuronDeviceIndexValueExists := datapoint.Attributes().Get(NeuronDeviceAttributeKey)
+
+				if neuronCoreIndexValueExists && neuronDeviceIndexValueExists {
+					neuronCoreInfo := NeuronCoreInfo{neuronCoreIndex: neuronCoreIndexValue.AsString(), neuronDeviceIndex: neuronDeviceIndexValue.AsString()}
+
+					currentValue, exists := d.memoryMetricValuesAggregator[neuronCoreInfo]
+					if exists {
+						d.memoryMetricValuesAggregator[neuronCoreInfo] = currentValue + datapoint.DoubleValue()
+					} else {
+						d.memoryMetricValuesAggregator[neuronCoreInfo] = datapoint.DoubleValue()
+					}
+				}
+			}
+		}
+	}
+}
+
+func (d *AwsNeuronMemoryMetricsAggregator) FlushAggregatedMemoryMetric() pmetric.Metric {
+	aggregatedMemoryMetric := pmetric.NewMetric()
+	aggregatedMemoryMetric.SetName(containerinsightscommon.NeuronCoreMemoryUtilizationTotal)
+	datapoints := aggregatedMemoryMetric.SetEmptySum().DataPoints()
+
+	for neuronCoreInfo, totalMemoryUsed := range d.memoryMetricValuesAggregator {
+		datapoint := datapoints.AppendEmpty()
+		datapoint.SetDoubleValue(totalMemoryUsed)
+		d.aggregatedMemoryMetricAttributes.CopyTo(datapoint.Attributes())
+
+		datapoint.Attributes().PutStr(NeuronCoreAttributeKey, neuronCoreInfo.neuronCoreIndex)
+		datapoint.Attributes().PutStr(NeuronDeviceAttributeKey, neuronCoreInfo.neuronDeviceIndex)
+		datapoint.SetTimestamp(d.metricTimestamp)
+	}
+
+	// Reset the aggregator
+	d.resetMemoryMetricAggregator()
+	return aggregatedMemoryMetric
+}
+
+func (d *AwsNeuronMemoryMetricsAggregator) resetMemoryMetricAggregator() {
+	d.memoryMetricValuesAggregator = map[NeuronCoreInfo]float64{}
```

Comment:
I think I left one out that is the runtime_tag, rest all will be the same for a scrape since they all are instance related info which would not change.